### ISSUE TITLE
Adding many files causes performance slowdown

### ISSUE THREAD

**Issue by jcoyne** [projecthydra-labs/curation_concerns#431]
It takes 1 minute to add a single file to the work after 120 files.
```
    15:28:26 attaching file: letz_01_0099_098.tif
    15:29:16 attaching file: letz_01_0101_100.tif
    15:30:06 attaching file: letz_01_0102_101.tif
    15:30:55 attaching file: letz_01_0103_102.tif
    15:31:46 attaching file: letz_01_0104_103.tif
    15:32:35 attaching file: letz_01_0105_104.tif
    15:33:24 attaching file: letz_01_0106_105.tif
    15:34:14 attaching file: letz_01_0107_106.tif
    15:35:05 attaching file: letz_01_0108_107.tif
    15:35:56 attaching file: letz_01_0109_108.tif
    15:36:48 attaching file: letz_01_0110_109.tif
    15:37:40 attaching file: letz_01_0111_110.tif
    15:38:33 attaching file: letz_01_0112_111.tif
    15:39:27 attaching file: letz_01_0113_112.tif
    15:40:22 attaching file: letz_01_0114_113.tif
    15:41:17 attaching file: letz_01_0115_114.tif
    15:42:13 attaching file: letz_01_0116_115.tif
    15:43:10 attaching file: letz_01_0117_116.tif
    15:44:06 attaching file: letz_01_0118_117.tif
    15:45:05 attaching file: letz_01_0119_118.tif
    15:46:03 attaching file: letz_01_0120_119.tif
    15:47:01 attaching file: letz_01_0121_120.tif
    15:48:01 attaching file: letz_01_0122_121.tif
    15:49:01 attaching file: letz_01_0123_122.tif
```

**Comment by tpendragon** [comment]
@jcoyne Where do we think this is happening? Indexing?

**Comment by jcoyne** [comment]
I'll be looking into it later today.

**Comment by tpendragon** [comment]
So I don't think this should be anything in AF::Aggregations. My guesses would be A) Parsing the increasing amount of triples (list_source will get to be a couple thousand triples pretty quick) B) Indexing (the list_source index should be pretty quick, but the work indexing is possibly slow)

**Comment by jcoyne** [comment]
The problem appears to be that it's doing too many loads from Fedora:

https://gist.github.com/jcoyne/b1b2953c9156558cb1f9

**Comment by jcoyne** [comment]
https://github.com/projecthydra-labs/activefedora-aggregation/pull/98 and https://github.com/projecthydra/active_fedora/pull/925 (merged) have helped a lot. Unfortunately, it's still exhibiting exponential growth.

**Comment by grosscol** [comment]
Has the work of @tpendragon and @jcoyne resolved this issue?

**Comment by jcoyne** [comment]
Some. Still waiting on ActiveTriples fix to get backported to 0.7.x

**Comment by jcoyne** [comment]
Closed with ActiveTriples 0.7.4 https://rubygems.org/gems/active-triples/versions/0.7.4
